### PR TITLE
Fixes For MinGW-64

### PR DESCRIPTION
**Pull request by ibroheem**
**Fixed issues leading to compilation error on MinGW-64 GCC.**

There's also the issue of linking problems arising from incomplete detection/adding WinAPI libraries.
This should be trivial to do with cmake.

### CODE REVIEW

File: minethd.cpp [hunk 1]
```diff
@@ -24,6 +24,32 @@
 #include <assert.h>
 #include <cmath>
 #include <chrono>
+/*
```
Comment (psychocrypt) [inline minethd.cpp]: Why do you add this license header heare?

File: minethd.cpp [hunk 2]
```diff
@@ -530,3 +560,446 @@ void minethd::double_work_main()
 	cryptonight_free_ctx(ctx0);
 	cryptonight_free_ctx(ctx1);
 }
+
```
Comment (psychocrypt) [inline minethd.cpp]: Where are this changes coming from?
Is this part of the fix or is it a mistake that this is addded here.

Comment (ibroheem) [inline minethd.cpp]: **Mistake!**
Probably (AStyle) Styling error/incosistiency leading to false +ve changes.
Nothing to do with the code.

Sorry!

File: minethd.cpp (hunk 1 above)
Comment (ibroheem) [inline minethd.cpp]: Nothing to do with the code.
**Sorry guys!**

**Comment by fireice-uk** [comment]
Holy cow this PR is a mess... @ibroheem I will close it since it can't be reviewed for changes. Please make sure only your commits are in a PR and you are not duplicating licence headers.

**Pull request closed without merging**